registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -228,114 +228,115 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeDerivatives (Eige
-NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives (Eigen::Matrix<double, 6, 1> &p, bool compute_hessian)
+NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives (Eigen::Matrix<double, 6, 1> &transform, bool compute_hessian)
 {
   // Simplified math for near 0 angles
   double cx, cy, cz, sx, sy, sz;
-  if (std::abs (p (3)) < 10e-5)
+  if (std::abs (transform (3)) < 10e-5)
   {
-    //p(3) = 0;
     cx = 1.0;
     sx = 0.0;
   }
   else
   {
-    cx = std::cos (p (3));
-    sx = sin (p (3));
+    cx = std::cos (transform (3));
+    sx = sin (transform (3));
   }
-  if (std::abs (p (4)) < 10e-5)
+  if (std::abs (transform (4)) < 10e-5)
   {
-    //p(4) = 0;
     cy = 1.0;
     sy = 0.0;
   }
   else
   {
-    cy = std::cos (p (4));
-    sy = sin (p (4));
+    cy = std::cos (transform (4));
+    sy = sin (transform (4));
   }
 
-  if (std::abs (p (5)) < 10e-5)
+  if (std::abs (transform (5)) < 10e-5)
   {
-    //p(5) = 0;
     cz = 1.0;
     sz = 0.0;
   }
   else
   {
-    cz = std::cos (p (5));
-    sz = sin (p (5));
+    cz = std::cos (transform (5));
+    sz = sin (transform (5));
   }
 
-  // Precomputed angular gradiant components. Letters correspond to Equation 6.19 [Magnusson 2009]
-  j_ang_a_ << (-sx * sz + cx * sy * cz), (-sx * cz - cx * sy * sz), (-cx * cy);
-  j_ang_b_ << (cx * sz + sx * sy * cz), (cx * cz - sx * sy * sz), (-sx * cy);
-  j_ang_c_ << (-sy * cz), sy * sz, cy;
-  j_ang_d_ << sx * cy * cz, (-sx * cy * sz), sx * sy;
-  j_ang_e_ << (-cx * cy * cz), cx * cy * sz, (-cx * sy);
-  j_ang_f_ << (-cy * sz), (-cy * cz), 0;
-  j_ang_g_ << (cx * cz - sx * sy * sz), (-cx * sz - sx * sy * cz), 0;
-  j_ang_h_ << (sx * cz + cx * sy * sz), (cx * sy * cz - sx * sz), 0;
+  // Precomputed angular gradient components. Letters correspond to Equation 6.19 [Magnusson 2009]
+  angular_jacobian_.setZero();
+  angular_jacobian_.row(0).noalias() = Eigen::Vector4d((-sx * sz + cx * sy * cz), (-sx * cz - cx * sy * sz), (-cx * cy), 1.0);  // a
+  angular_jacobian_.row(1).noalias() = Eigen::Vector4d((cx * sz + sx * sy * cz), (cx * cz - sx * sy * sz), (-sx * cy), 1.0);    // b
+  angular_jacobian_.row(2).noalias() = Eigen::Vector4d((-sy * cz), sy * sz, cy, 1.0);                                           // c
+  angular_jacobian_.row(3).noalias() = Eigen::Vector4d(sx * cy * cz, (-sx * cy * sz), sx * sy, 1.0);                            // d
+  angular_jacobian_.row(4).noalias() = Eigen::Vector4d((-cx * cy * cz), cx * cy * sz, (-cx * sy), 1.0);                         // e
+  angular_jacobian_.row(5).noalias() = Eigen::Vector4d((-cy * sz), (-cy * cz), 0, 1.0);                                         // f
+  angular_jacobian_.row(6).noalias() = Eigen::Vector4d((cx * cz - sx * sy * sz), (-cx * sz - sx * sy * cz), 0, 1.0);            // g
+  angular_jacobian_.row(7).noalias() = Eigen::Vector4d((sx * cz + cx * sy * sz), (cx * sy * cz - sx * sz), 0, 1.0);             // h
```

Comment:
Double check if this can all be replaced by the `operator<<` that the matrix class exposes

Reply:
When I did profiling, I observed this part can be

Reply:
When run profiler on this code a few years ago, I found that this part made a bit slow by the use of operator<<. I think it would be better to stick to the current code if it's acceptable. (Recent compilers may generate optimized codes with operator<< though)